starlette/staticfiles.py

```diff
@@ -73,19 +76,19 @@ def __init__(self, scope, path, check_directory=None):
         if not (stat.S_ISDIR(stat_result.st_mode) or stat.S_ISLNK(stat_result.st_mode)):
             raise RuntimeError("StaticFiles path '%s' is not a directory." % directory)
 
-    async def __call__(self, receive, send):
+    async def __call__(self, receive: Receive, send: Send) -> None:
         if self.check_directory is not None:
             await self.check_directory_configured_correctly()
 
         try:
             stat_result = await aio_stat(self.path)
         except FileNotFoundError:
-            response = PlainTextResponse("Not Found", status_code=404)
+            response = PlainTextResponse("Not Found", status_code=404)  # type: Response
         else:
             mode = stat_result.st_mode
+
+            response = FileResponse(self.path, stat_result=stat_result)
```

Comment:
I'd prefer that we don't move this logic around. It's better to keep this in the `else` block and only instantiate one or the other case.

Reply:
Perfect!